code/game/gamemodes/wizard/soulstone.dm

```diff
@@ -200,17 +214,35 @@
 		..()
 
 /obj/item/soulstone/attack_self(mob/living/user)
-	if(!in_range(src, user))
+	var/mob/living/simple_animal/shade/S = locate(/mob/living/simple_animal/shade) in contents
+	if(!in_range(src, user) || !S)
 		return
 
-	if(!can_use(user))
-		user.Weaken(10 SECONDS)
-		user.emote("scream")
-		to_chat(user, "<span class='userdanger'>Your body is wracked with debilitating pain!</span>")
+	if(can_use(user))
+		release_shades(user)
 		return
 
-	release_shades(user)
-	return
+	if(!user.mind.isholy)
+		to_chat(user, "<span class='notice'>The shard feels too tough to shatter, you are not holy enough to free its captive!</span>")
+		return
+
+	if(!do_after_once(user, 10 SECONDS, FALSE, src) || !held_body)
+		return
+
+	user.visible_message("[user] shatters the soulstone apart! Releasing [held_body] from their prison!", "You shatter the soulstone holding [held_body], binding them free!", "You hear something shatter with a ghastly crack.")
+	if(ismob(held_body))
+		var/mob/M = held_body
+		M.key = S.key
+	else if(istype(held_body, /obj/item/organ/internal/brain))
+		var/obj/item/organ/internal/brain/B = held_body
+		B.brainmob.key = S.key
```

Comment:
Unless I'm missing something, shouldn't you only be able to shard mobs? Why is there a check for the held body being a brain?

Reply:
I believe this was written like this to absorb brains into shards, this was the fix for it. As well as head organs that contain brains

Reply:
Marking as resolved unless any other recommendations.